### PR TITLE
It would seem our documentation is inaccurate BA-5828

### PR DESCRIPTION
**Pull request by geoffjentry**

### CODE REVIEW

File: docs/api/RESTAPI.md [hunk 1]
```diff
@@ -283,7 +283,7 @@ Query for workflows which match various criteria. When a combination of criteria
 |Type|Name|Description|Schema|Default|
 |---|---|---|---|---|
 |**Path**|**version**  <br>*required*|Cromwell API Version|string|`"v1"`|
-|**Query**|**additionalQueryResultFields**  <br>*optional*|Currently only 'labels' is a valid value here. Use it to include a list of labels with each result.|< string > array(multi)||
+|**Query**|**additionalQueryResultFields**  <br>*optional*|Currently only 'labels' and `parentworkflowId` are valid values here. Use it to include a list of labels and/or parent workflow IDs with each result.|< string > array(multi)||
```
Comment (cjllanwarne) [inline docs/api/RESTAPI.md]: Is that capitalization really correct for `parentworkflowId `...? 😨

Comment (cjllanwarne) [reply docs/api/RESTAPI.md]: In the other file this is written as `parentWorkflowId `

Comment (mcovarr) [inline docs/api/RESTAPI.md]:
```suggestion
|**Query**|**additionalQueryResultFields**  <br>*optional*|Currently only 'labels' and `parentWorkflowId` are valid values here. Use these to include a list of labels and/or parent workflow ID with each result.|< string > array(multi)||
```

File: engine/src/main/resources/swagger/cromwell.yaml [hunk 2]
```diff
@@ -436,7 +436,7 @@ paths:
             type: string
           collectionFormat: multi
           description: >
-            Currently only 'labels' is a valid value here. Use it to include a list of labels with each result.
+            Currently only 'labels' and `parentWorkflowId` are valid values here. Use it to include a list of labels and/or parent workflow IDs with each result.
```
Comment (mcovarr) [inline engine/src/main/resources/swagger/cromwell.yaml]:
```suggestion
            Currently only 'labels' and `parentWorkflowId` are valid values here. Use these to include a list of labels and/or parent workflow ID with each result.
```

**Comment by cjllanwarne** [comment]
Worth checking this but I have a feeling that this was correct as it was before. I because `parentWorkflowId` is now returned in all cases anyway, so adding this field would not actually change the response at all.

File: docs/api/RESTAPI.md (hunk 1 above)
Comment (geoffjentry) [reply docs/api/RESTAPI.md]: @cjllanwarne Yeah duh. I just did a global search for text files which had this, forgot about the autogeneration

**Comment by geoffjentry** [comment]
@cjllanwarne If that's the case then it'd appear that not once, but at least twice now people have fiddled with the behavior of this in the query endpoint without updating the docs and/or all of the relevant code :)

from `case object AdditionalQueryResultFields ...`
`val allowedValues = Seq(WorkflowMetadataKeys.Labels, WorkflowMetadataKeys.ParentWorkflowId)`

**Comment by cjllanwarne** [comment]
I'm pretty sure I'm the last to touch this. IIRC (but nb: emphasis on the *if*, in this case):

* The docs used to say "put any field in here and we'll return it in your query results". That was never true.
* Because parent workflow ID is now in the summary table, there's no longer any cost to always returning it, so we do, alongside the other summarized fields.
* I probably didn't want to break anyone who'd hard-wired `parentWorkflowId` into their queries, so didn't remove the "validity" of that argument, even if it's no longer "active".

**Comment by cjllanwarne** [comment]
For reference: https://github.com/broadinstitute/cromwell/pull/4831

**Comment by geoffjentry** [comment]
@cjllanwarne that makes sense. In that case I think the thing to do here would be to revert the doc change and add a comment to the offending line in the code so that it doesn't confuse a future traveler

**Comment by geoffjentry** [comment]
@cjllanwarne @mcovarr now radically different

**Comment by mcovarr** [comment]
Apparently you broke a Spark unit test with this.

**Comment by geoffjentry** [comment]
Comments are tricky like that

**Pull request merged**